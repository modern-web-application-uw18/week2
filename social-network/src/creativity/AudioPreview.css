.AudioPreview {
    display: flex;
    flex-wrap: nowrap;
    color: gray;
    font-weight: 500;
    height: 20px;
    width: 224px;
    padding-bottom: 10px;
}

.Article-audio-text {
    font-size: 0.75em;
    text-align: left;
    padding-right: 75px;
}

.Article-audio-image {
    padding-right: 10px;
}

.Article-star-container {
    height: 20px;
    width: 20px;
}

.Article-star-container img {
    height: auto;
    width: 65%;
}
