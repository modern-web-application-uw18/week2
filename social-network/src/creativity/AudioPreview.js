import React, { Component } from 'react';
import PropTypes from 'prop-types';
import './AudioPreview.css';
import speaker from '../img/icon-speaker.png';
import star from '../img/icon-star.svg';

class AudioPreview extends Component {

    render() {

        if (this.props.hasAudioAvailable) {
            return (
                <div className="AudioPreview">
                    <div className="Article-audio-image">
                        <img src={speaker} alt="Play Audio" />
                    </div>
                    <div className="Article-audio-text">Audio Available</div>
                    {this.props.memberPreview ? (
                        <div className="Article-star-container">
                            <img src={star} alt="" />
                        </div>
                    ) : null}
                </div>
            )
        } else if (this.props.memberPreview) {
            return (
                <div className="AudioPreview">
                    <div className="Article-star-container">
                        <img src={star} alt="" />
                    </div>
                    <div className="Article-audio-text">Member preview</div>
                </div>
            )
        } else {
            return null;
        }
    }
}


AudioPreview.propTypes = {
    hasAudioAvailable: PropTypes.bool.isRequired,
    memberPreview: PropTypes.bool.isRequired
};


AudioPreview.defaultProps = {
    hasAudioAvailable: false,
    memberPreview: false
};

export default AudioPreview;