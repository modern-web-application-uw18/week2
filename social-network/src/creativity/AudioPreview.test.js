import React from 'react';
import ReactDOM from 'react-dom';
import TestRenderer from 'react-test-renderer';
import AudioPreview from './AudioPreview';

describe('ArticleAudio', () => {
    it('renders without crashing', () => {
        const articleAudio = document.createElement('ArticleAudio');
        ReactDOM.render(<AudioPreview
            hasAudioAvailable={true}
            memberPreview={false} />, articleAudio);
        ReactDOM.unmountComponentAtNode(articleAudio);
    });

    it('should render with hasAudioAvailable, memberPreview', () => {
        const component = TestRenderer.create(
            <AudioPreview hasAudioAvailable={true} memberPreview={false} />);
        expect(component.toJSON()).toMatchSnapshot();
    });
});
