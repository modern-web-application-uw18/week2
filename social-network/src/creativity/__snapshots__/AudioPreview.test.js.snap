// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleAudio should render with hasAudioAvailable, memberPreview 1`] = `
<div
  className="AudioPreview"
>
  <div
    className="Article-audio-image"
  >
    <img
      alt="Play Audio"
      src="icon-speaker.png"
    />
  </div>
  <div
    className="Article-audio-text"
  >
    Audio Available
  </div>
</div>
`;
